src/Workspaces/Core/Portable/Execution/ChecksumTreeCollection.TreeNodes.cs

```diff
@@ -179,6 +247,59 @@ public IChecksumTreeNode GetOrCreateSubTreeNode<TKey>(TKey key)
                 return GetOrCreateChecksumObjectAsync(key, value, kind, valueGetterAsync, cancellationToken);
             }
 
+            protected void AppendChecksumObjects(
+                ImmutableDictionary<Checksum, ChecksumObject>.Builder builder,
+                HashSet<Checksum> checksums,
+                int itemCount,
+                IEnumerable<Checksum> itemChecksums,
+                Func<Checksum, ChecksumObject> itemGetter, CancellationToken cancellationToken)
+            {
```

Comment:
one set is the set user asked to find, and one set is the checksums this subtree owns.

I look over checksum from either one set (smaller set) and check whether both set has it.